tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp

```diff
@@ -21,20 +21,56 @@ struct LayoutMapper<std::experimental::layout_right> {
   using type = Kokkos::LayoutRight;
 };
 
+template<class T>
+constexpr void static_assert_complex_alignment(const T* /* p */)
+{
+  using kc_t = Kokkos::complex<typename T::value_type>;
+  static_assert(alignof(T) == alignof(kc_t),
```

Comment:
Please see comment above.  Users should still be able to create a `Kokkos::View` from complex data if they align their data manually.  Doing a run-time test would, of course, be a bit harder on device, but Kokkos users generally aren't letting `std::complex` data exist in device code.